splinterd/src/daemon/oauth.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright 2018-2021 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Provides an builder for OAuthConfig from string values.
+
+use splinter::error::InvalidStateError;
+use splinter::rest_api::OAuthConfig;
+use splinter::store::StoreFactory;
+
+enum RunnableOAuthProvider {
+    Azure {
+        oauth_openid_url: String,
+    },
+    Github,
+    Google,
+    OpenId {
+        oauth_openid_url: String,
+        auth_params: Option<Vec<(String, String)>>,
+        scopes: Option<Vec<String>>,
+    },
+}
+
+/// A configured, but not fully runnable OAuth configuration.
+pub struct RunnableOAuthConfig {
```

Comment:
"Runnable" doesn't really make sense in the context of configuration--do we transition to an object that owns threads here?

Reply:
That’s a good point. The store factory isn’t ready yet, but the config is validated. Maybe `ValidatedOAuthConfig` or `PeparedOAuthConfig`?

Reply:
`PartialOAuthConfig`?

Reply:
I've renamed this to `PartialOAuthConfig`